types/wlr_output_swapchain.c

```diff
@@ -0,0 +1,236 @@
+#include <assert.h>
+#include <drm_fourcc.h>
+#include <stdlib.h>
+#include <wlr/backend.h>
+#include <wlr/interfaces/wlr_output.h>
+#include <wlr/render/wlr_renderer.h>
+#include <wlr/types/wlr_output.h>
+#include <wlr/types/wlr_output_swapchain.h>
+#include <wlr/util/log.h>
+#include "backend/backend.h"
+#include "render/allocator/allocator.h"
+#include "render/drm_format_set.h"
+#include "render/swapchain.h"
+#include "render/wlr_renderer.h"
+
+struct wlr_output_swapchain_manager *wlr_output_swapchain_manager_create(
+		struct wlr_backend *backend) {
+	struct wlr_output_swapchain_manager *manager = calloc(1, sizeof(*manager));
+	if (manager == NULL) {
+		return NULL;
+	}
+
+	manager->renderer = wlr_renderer_autocreate(backend);
```

Comment:
Should the helper "own" the renderer? `wlr_backend_get_renderer` seems more appropriate for external renderers.

Reply:
I want to get rid of `wlr_backend_get_renderer` eventually, because the backend should be completely separate from the renderer.